app/controllers/posts_controller.rb

```diff
@@ -5,6 +5,37 @@ def index
   end
 
   def show
-    @posts = Post.find(params[:id])
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:id])
+  end
+
+  def new
+    @user = User.find(params[:user_id])
+    @post = Post.new
+  end
+
+  def create
+    # @user = User.find(params[:user_id])
+    @post = Post.new(post_params)
+    @post.author = current_user
+    @post.comment_counter = 0
+    @post.like_counter = 0
```

Comment:
- [ ] Please remove the commented line of code above as it doesn't make your code look clean 👍 

- [ ] [OPTIONAL] Please try to set a default value of 0 for the `comments_counter` and `likes_counter` at the database level so you don't have to worry about it in your code like you have done here 👍

Reply:
Thanks a lot for this suggestion, I will do well to implement them👍.